<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction to MethEvolSIM</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">

div.csl-bib-body { }
div.csl-entry {
clear: both;
margin-bottom: 0em;
}
.hanging div.csl-entry {
margin-left:2em;
text-indent:-2em;
}
div.csl-left-margin {
min-width:2em;
float:left;
}
div.csl-right-inline {
margin-left:2em;
padding-left:1em;
}
div.csl-indent {
margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to MethEvolSIM</h1>


<div id="TOC">
<ul>
<li><a href="#a-first-example" id="toc-a-first-example">A first
example</a></li>
<li><a href="#model-parameters" id="toc-model-parameters">Model
parameters</a>
<ul>
<li><a href="#initial-methylation-state-distribution" id="toc-initial-methylation-state-distribution">Initial methylation
state distribution</a></li>
<li><a href="#dna-methylation-evolution-processes" id="toc-dna-methylation-evolution-processes">DNA methylation evolution
processes</a>
<ul>
<li><a href="#island-wide-events-iwes" id="toc-island-wide-events-iwes">Island-wide events (IWEs)</a></li>
<li><a href="#single-site-events-sses" id="toc-single-site-events-sses">Single-site events (SSEs)</a></li>
</ul></li>
</ul></li>
<li><a href="#simulation-setup" id="toc-simulation-setup">Simulation
Setup</a>
<ul>
<li><a href="#structural-distribution-of-cpg-sites" id="toc-structural-distribution-of-cpg-sites">Structural distribution of
CpG sites</a></li>
<li><a href="#initial-methylation-state-distribution-1" id="toc-initial-methylation-state-distribution-1">Initial methylation
state distribution</a></li>
<li><a href="#parameter-values" id="toc-parameter-values">Parameter
values</a></li>
</ul></li>
<li><a href="#simulation-of-dna-methylation-data" id="toc-simulation-of-dna-methylation-data">Simulation of DNA
Methylation Data</a>
<ul>
<li><a href="#simulation-of-initial-data" id="toc-simulation-of-initial-data">Simulation of Initial Data</a>
<ul>
<li><a href="#exploration-of-initial-data" id="toc-exploration-of-initial-data">Exploration of Initial
Data</a></li>
</ul></li>
<li><a href="#simulation-of-data-evolution" id="toc-simulation-of-data-evolution">Simulation of Data
Evolution</a></li>
</ul></li>
<li><a href="#references" id="toc-references">References</a></li>
</ul>
</div>

<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(MethEvolSIM)</span></code></pre></div>
<p>MethEvolSIM is an R package for simulating DNA methylation dynamics
on different genomic structures along a given tree. It allows simulating
methylation and demethylation events at loci within two different types
of genomic structures: regions with low methylation or with high
methylation. The data can be interpreted as:</p>
<p>1- CpG sites inside or outside CpG islands or</p>
<p>2- any type of methylation site in regions of higher or lower
methylation or</p>
<p>3- CpG islands inside or outside genomic archipelagos (regions with
many CpG islands).</p>
<p>In this document, unless specified otherwise, we use the first
interpretation.</p>
<div id="a-first-example" class="section level1">
<h1>A first example</h1>
<p>As a simple example with default parameter values we show a
simulation of evolution of a simplified genomic structure on a tree:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># Define the tree:</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="st">&quot;((a:1, b:1):2, c:2);&quot;</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co"># Define the genomic structure. Here 1 island (&quot;U&quot;) containing 100 CpGs </span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># surrounded by 2 non-island structures (&quot;M&quot;) each containing 10 CpGs</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">100</span>, <span class="dv">10</span>),</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>                      <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>))</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co"># Simulate 1 replicate of data at the tree tips</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>output <span class="ot">&lt;-</span> <span class="fu">simulate_evolData</span>(<span class="at">infoStr =</span> infoStr, <span class="at">tree =</span> tree)</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; Using default parameter values</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2);</span></span></code></pre></div>
<p>In the simulation output 0 stands for unmethylated, 1 for methylated
and 0.5 for partially methylated or hemimethylated.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># In replicate 1, tip 2: name of the tip of the tree</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>output<span class="sc">$</span>data[[<span class="dv">1</span>]][[<span class="dv">2</span>]]<span class="sc">$</span>name</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;b&quot;</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co"># In replicate 1, tip 2: sequence of methylation states of left non-island structure</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>output<span class="sc">$</span>data[[<span class="dv">1</span>]][[<span class="dv">2</span>]]<span class="sc">$</span>seq[[<span class="dv">1</span>]]</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt;  [1] 0.0 0.0 0.0 0.0 0.0 0.0 0.5 0.0 0.0 0.0</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co"># In replicate 1, tip 2: sequence of methylation states of island structure</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>output<span class="sc">$</span>data[[<span class="dv">1</span>]][[<span class="dv">2</span>]]<span class="sc">$</span>seq[[<span class="dv">2</span>]]</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt;   [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt;  [38] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt;  [75] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co"># In replicate 1, tip 2: sequence of methylation states of right non-island structure</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>output<span class="sc">$</span>data[[<span class="dv">1</span>]][[<span class="dv">2</span>]]<span class="sc">$</span>seq[[<span class="dv">3</span>]]</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt;  [1] 0 1 1 1 1 1 1 1 1 1</span></span></code></pre></div>
</div>
<div id="model-parameters" class="section level1">
<h1>Model parameters</h1>
<div id="initial-methylation-state-distribution" class="section level2">
<h2>Initial methylation state distribution</h2>
<p>Structures are categorized as CpG islands or non-islands
(alternatively regions with low/high methylation or
archipelagos/non-archipelagos). In the root of the genealogy we assume
that each structure samples an equilibrium probability triple <span class="math inline">\((\pi_u,\pi_p, \pi_m)\)</span> for the methylation
states: unmethylated, partially-methylated and methylated, respectively.
Current parameter values favor prevalent homogeneity in methylation
levels within structures, most CpG islands predominantly unmethylated
and a small proportion predominantly methylated, and most non-islands
predominantly methylated and a small proportion predominantly
unmethylated. The sampling of the equilibrium probability triple is done
using two distinct Beta distributions to sample the corresponding
methylation frequencies.</p>
<p><strong>Island parameterization: alpha_pI, beta_pI, alpha_mI,
beta_mI</strong></p>
<p>MethEvolSIM samples <span class="math inline">\(\pi_p\)</span> from
the first Beta distribution. Current parameterization uses as initial
parameter values a relatively small <span class="math inline">\(\alpha_{I_p}\)</span> and relatively large <span class="math inline">\(\beta_{I_p}\)</span> to encourage minor assignment
of partially methylated states.</p>
<p>The second Beta distribution, scaled by <span class="math inline">\(1
- \pi_p\)</span>, corresponds to the proportion of methylated and
unmethylated states. Currently both parameter values (<span class="math inline">\(\alpha_{I_m}\)</span> and <span class="math inline">\(\beta_{I_m}\)</span>) set to small values while
<span class="math inline">\({\alpha_{I_m} &lt; \beta_{I_m}}\)</span> to
favour shifting the stochastic choice toward homogeneous states with a
higher proportion of predominantly unmethylated islands, and assigned to
<span class="math inline">\(\pi_m\)</span>. Finally, we set <span class="math inline">\(\pi_u\)</span> as <span class="math inline">\(1 -
\pi_p - \pi_m\)</span>.</p>
<p><strong>Non-island parameterization: alpha_pNI, beta_pNI, alpha_mNI,
beta_mNI</strong></p>
<p>Following the same rationale as for island parameterization, each
non-island structure samples a common equilibrium probability triple
from two Beta distributions with parameters <span class="math inline">\(\alpha_{NI_p}\)</span>, <span class="math inline">\(\beta_{NI_p}\)</span>, <span class="math inline">\(\alpha_{NI_m}\)</span> and <span class="math inline">\(\beta_{NI_m}\)</span>. The parameter values are
currently set to favor the characteristic high overall methylation
level.</p>
<p><strong>Parameter ranges</strong> The alpha and beta parameters of a
Beta distribution must be greater than 0, that is in (0, <span class="math inline">\(\infty\)</span>).</p>
</div>
<div id="dna-methylation-evolution-processes" class="section level2">
<h2>DNA methylation evolution processes</h2>
<p>DNA methylation state evolution happens by two structure-specific
events: CpG single site events (SSEs) and CpG island wide events
(IWEs).</p>
<div id="island-wide-events-iwes" class="section level3">
<h3>Island-wide events (IWEs)</h3>
<p>IWEs change the island methylation probabilities and some of the CpG
sites in the CpG island can simultaneously change their state at the
same time. Different CpG islands, however, are assumed to evolve
independently of each other. IWEs occur per CpG island at a rate denoted
by <span class="math inline">\(\mu\)</span>. Consequently, over a branch
of length <span class="math inline">\(l\)</span>, the expected number of
IWEs per island is given by <span class="math inline">\(\mu \cdot
l\)</span>.</p>
<p><strong>IWE parameterization: mu. Parameter range:</strong> Should be
a small number, but it can take any value from 0, that is in <span class="math inline">\([0, \infty)\)</span></p>
</div>
<div id="single-site-events-sses" class="section level3">
<h3>Single-site events (SSEs)</h3>
<p>In addition to IWEs we allow for single site events (SSEs), which
change the methylation states of single CpG sites within and between CpG
islands. In our model SSEs account for two dynamics: independent of each
other (SSEi), and assuming correlations between adjacent sites
(SSEc).</p>
<p>As SSEc transition probabilities are dependent on adjacent sites we
use Gillespie’s <span class="math inline">\(\tau\)</span>-leap
approximation <span class="citation">(Gillespie 2001)</span> to allow
for CpG sites to o update their neighbouring state and rates of change
after short time steps. Therefore, we first discretize the branch
intervals between IWEs in time intervals of default length <span class="math inline">\(dt =0.01\)</span>. For each CpG position and each
time interval the rate matrix for the transitions between the states
unmethylated, partially methylated and methylated results from the
addition of the rate matrices of the two types of SSE events, so that
<span class="math inline">\(\mathbb{E}(R_i + R_c)=1\)</span>, <span class="math inline">\(R_i\)</span> and <span class="math inline">\(R_c\)</span> representing the rate factors of
independent and correlated processes. The independent process samples
the new methylation state from the structure’s equilibrium frequencies
and the correlated process copies the left or right neighbor state with
equal probabilities. The proportion of each of the two SSE types is
determined by the model parameter <span class="math inline">\(\iota\)</span>.</p>
<p>Different positions <span class="math inline">\(j\)</span> have
different SSEi rates <span class="math inline">\(R_{i,j}\)</span> coming
from a discretized gamma distribution with 3 categories, expected value
<span class="math inline">\(\iota\)</span> and shape parameters <span class="math inline">\(\alpha_{R_{i}}\)</span> and <span class="math inline">\({\beta_{R_{i}}}=\frac{\alpha_{R_{i}}}{\iota}\)</span>.
The probability to be in each respective rate category is 1/3. For the
rate of collaborative SSEs <span class="math inline">\(R_c =
1-\iota\)</span>, the probability of considering left of right neighbour
is equal.</p>
<p><strong>SSE process parameterization: iota, alpha_Ri</strong>
<strong>Parameter ranges</strong> The alpha parameter of a Gamma
distribution must be greater than 0, that is (0, <span class="math inline">\(\infty\)</span>). Note that due to numerical
limitations, the smallest value used by MethEvolSIM is 0.01. Iota must
be between 0 (non-included) and 1, that is in (0,1].</p>
</div>
</div>
</div>
<div id="simulation-setup" class="section level1">
<h1>Simulation Setup</h1>
<div id="structural-distribution-of-cpg-sites" class="section level2">
<h2>Structural distribution of CpG sites</h2>
<p>The initial spatial distribution of CpG sites in islands and
non-island structures is given by the user through a data frame with 2
columns named “n” and “globalState” , which encode the structural
information. Each row provides the information of a different genomic
structure.</p>
<ul>
<li><p><strong>n</strong> number of positions for each structure
(e.g. number of CpG sites).</p></li>
<li><p><strong>globalState</strong> encodes island and non-island
structures with the character value “U” for unmethylated regions
(e.g. CpG islands) and “M” for methylated regions (e.g. non-CpG
islands).</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Example 3 structures of length 100 each:</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co"># one non-island containing 100 CpG sites,</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co"># one island containing 100 CpG sites,</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co"># and one non-island containing 100 CpG sites</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>                        <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>))</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>infoStr</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt;     n globalState</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 1 100           M</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 2 100           U</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; 3 100           M</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Example for 2 islands: first of length 1, second of length 15</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">15</span>),</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>                      <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;U&quot;</span>, <span class="st">&quot;U&quot;</span>))</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>infoStr</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;    n globalState</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 1  1           U</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 2 15           U</span></span></code></pre></div>
</div>
<div id="initial-methylation-state-distribution-1" class="section level2">
<h2>Initial methylation state distribution</h2>
<p>The initial methylation state distribution assigns a triple of
methylation equilibrium frequencies() <span class="math inline">\(\pi_u\)</span> for unmethylated, <span class="math inline">\(\pi_p\)</span> for partially-methylated, and <span class="math inline">\(\pi_m\)</span> for methylated) to each of the
simulated genomic structures. This assignation can be done in two
ways:</p>
<p>1 - Sampled as specified in the section “Model parameters: Initial
methylation state distribution” when the given <code>infoStr</code> only
contains <code>n</code> and <code>globalState</code> columns.</p>
<p>2 - Determined by the user providing in <code>infoStr</code>
additional columns: <code>u_eqFreq</code> for <span class="math inline">\(\pi_u\)</span>, <code>p_eqFreq</code> for <span class="math inline">\(\pi_p\)</span> and <code>m_eqFreq</code> for <span class="math inline">\(\pi_m\)</span>. Note that when given, the triple
of frequency values for each of the structures needs to sum up to 1, so
that for each row <span class="math inline">\(i\)</span>, <span class="math inline">\(\pi_u(i)+\pi_p(i)+\pi_m(i)=1\)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># Example 3 structures of length 100 with customized initial methylation state</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co"># equilibrium frequencies</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>                      <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>),</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>                      <span class="at">u_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>),</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>                      <span class="at">p_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>                      <span class="at">m_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>infoStr</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt;     n globalState u_eqFreq p_eqFreq m_eqFreq</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; 1 100           M      0.1      0.1      0.8</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; 2 100           U      0.8      0.1      0.1</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; 3 100           M      0.1      0.1      0.8</span></span></code></pre></div>
</div>
<div id="parameter-values" class="section level2">
<h2>Parameter values</h2>
<p>The default parameter values used by MethEvolSIM can be obtained
using the function <code>get_parameterValues()</code>, which returns a
data frame with the columns named as specified in the section “Model
parameters”:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>default_paramValues <span class="ot">&lt;-</span> <span class="fu">get_parameterValues</span>()</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>default_paramValues</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt;   alpha_pI beta_pI alpha_mI beta_mI alpha_pNI beta_pNI alpha_mNI beta_mNI  mu</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; 1      0.1       1      0.1     0.5       0.1        1       0.5      0.1 0.1</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt;   alpha_Ri iota         Ri1         Ri2       Ri3</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; 1      0.1  0.3 2.80506e-06 0.005770752 0.8942264</span></span></code></pre></div>
<p>Additionally, it returns the three rates of change for SSEi
process,<code>Ri1</code>, <code>Ri2</code>, and <code>Ri3</code> (see
section “Model Parameters: Single-site events (SSEs)”).</p>
<p>The section “Model parameters” provides further information about
each of the parameters, including the range of possible values for each
parameter. To simulate data with a different parameterization, the new
value of the chosen parameter(s) can be modified in the data frame, but
for the cases of <code>Ri1</code>, <code>Ri2</code>, and
<code>Ri3</code>, which are modified through <code>iota</code> and
<code>alpha_Ri</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># Example: modification of parameter iota to value 0.2</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>default_paramValues<span class="sc">$</span>iota <span class="ot">&lt;-</span> <span class="fl">0.2</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>default_paramValues</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt;   alpha_pI beta_pI alpha_mI beta_mI alpha_pNI beta_pNI alpha_mNI beta_mNI  mu</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; 1      0.1       1      0.1     0.5       0.1        1       0.5      0.1 0.1</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt;   alpha_Ri iota         Ri1         Ri2       Ri3</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; 1      0.1  0.2 2.80506e-06 0.005770752 0.8942264</span></span></code></pre></div>
<p>Additional details regarding the <code>get_parameterValues()</code>
function can be found in its documentation.</p>
</div>
</div>
<div id="simulation-of-dna-methylation-data" class="section level1">
<h1>Simulation of DNA Methylation Data</h1>
<p>DNA Methylation Data is simulated using three R6 classes <span class="citation">(Chang 2022)</span>:
<code>singleStructureGenerator</code>,
<code>combiStructureGenerator</code>, and
<code>treeMultiRegionSimulator</code>. Instances of the
<code>singleStructureGenerator</code> class represent individual genomic
structures (e.g. islands or non-islands) and are encapsulated within
instances of the <code>combiStructureGenerator</code> class, which, in
turn, represent the simulated data structure (e.g. a sequence consisting
of a non-island containing 100 CpGs, an island containing 100 CpGs, and
a non-island containing 100 CpGs). Finally, instances of the
<code>treeMultiRegionSimulator</code> class contain a
<code>combiStructureGenerator</code> object at each node of a given
tree, including its root and leaves.</p>
<p>Generally, simulations are conducted as explained in subsection
“Simulation of Data Evolution”, with a documented user-interface
function explaining the output structure:
<code>simulate_evolData()</code>. Alternativelly, the user can simulate
initial data so that the root of the tree is fixed as explained in
subsection “Simulation of Initial Data”. The latter option generates
objects of class <code>combiStructureGenerator</code> containing objects
of class <code>singleStructureGenerator</code>. In the subsection
“Simulation of Initial Data: Exploration of Initial Data” an example to
interact with the classes is given, while further information about the
attributes and methods of each class is provided in their
documentation.</p>
<div id="simulation-of-initial-data" class="section level2">
<h2>Simulation of Initial Data</h2>
<p>The initial data can be generated as an object of class
<code>combiStructureGenerator</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Example with customized initial methylation frequencies and customized parameter values</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>custom_infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                             <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>),</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>                             <span class="at">u_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>),</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>                             <span class="at">p_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                             <span class="at">m_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>custom_params <span class="ot">&lt;-</span> <span class="fu">get_parameterValues</span>()</span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>custom_params<span class="sc">$</span>mu <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>initial_customD <span class="ot">&lt;-</span> <span class="fu">simulate_initialData</span>(<span class="at">infoStr =</span> custom_infoStr, <span class="at">params =</span> custom_params)</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="co"># Returns customized parameters and simulated data</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>initial_customD<span class="sc">$</span>params</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="co">#&gt;   alpha_pI beta_pI alpha_mI beta_mI alpha_pNI beta_pNI alpha_mNI beta_mNI    mu</span></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a><span class="co">#&gt; 1      0.1       1      0.1     0.5       0.1        1       0.5      0.1 0.005</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a><span class="co">#&gt;   alpha_Ri iota         Ri1         Ri2       Ri3</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="co">#&gt; 1      0.1  0.3 2.80506e-06 0.005770752 0.8942264</span></span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>initial_customD<span class="sc">$</span>data</span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a><span class="co">#&gt; &lt;combiStructureGenerator&gt;</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="co">#&gt;   Public:</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a><span class="co">#&gt;     add_offspring_index: function (i) </span></span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a><span class="co">#&gt;     branch_evol: function (branch_length, dt, testing = FALSE) </span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a><span class="co">#&gt;     clone: function (deep = FALSE) </span></span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a><span class="co">#&gt;     copy: function () </span></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a><span class="co">#&gt;     get_IWE_events: function () </span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a><span class="co">#&gt;     get_island_index: function () </span></span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a><span class="co">#&gt;     get_island_number: function () </span></span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a><span class="co">#&gt;     get_mu: function () </span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a><span class="co">#&gt;     get_name: function () </span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a><span class="co">#&gt;     get_offspring_index: function () </span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a><span class="co">#&gt;     get_own_index: function () </span></span>
<span id="cb9-30"><a href="#cb9-30" tabindex="-1"></a><span class="co">#&gt;     get_parent_index: function () </span></span>
<span id="cb9-31"><a href="#cb9-31" tabindex="-1"></a><span class="co">#&gt;     get_singleStr: function (i) </span></span>
<span id="cb9-32"><a href="#cb9-32" tabindex="-1"></a><span class="co">#&gt;     get_singleStr_number: function () </span></span>
<span id="cb9-33"><a href="#cb9-33" tabindex="-1"></a><span class="co">#&gt;     initialize: function (infoStr, params = NULL, testing = FALSE) </span></span>
<span id="cb9-34"><a href="#cb9-34" tabindex="-1"></a><span class="co">#&gt;     set_IWE_events: function (a) </span></span>
<span id="cb9-35"><a href="#cb9-35" tabindex="-1"></a><span class="co">#&gt;     set_name: function (a) </span></span>
<span id="cb9-36"><a href="#cb9-36" tabindex="-1"></a><span class="co">#&gt;     set_offspring_index: function (i) </span></span>
<span id="cb9-37"><a href="#cb9-37" tabindex="-1"></a><span class="co">#&gt;     set_own_index: function (i) </span></span>
<span id="cb9-38"><a href="#cb9-38" tabindex="-1"></a><span class="co">#&gt;     set_parent_index: function (i) </span></span>
<span id="cb9-39"><a href="#cb9-39" tabindex="-1"></a><span class="co">#&gt;     set_singleStr: function (singStrList) </span></span>
<span id="cb9-40"><a href="#cb9-40" tabindex="-1"></a><span class="co">#&gt;   Private:</span></span>
<span id="cb9-41"><a href="#cb9-41" tabindex="-1"></a><span class="co">#&gt;     IWE_events: NULL</span></span>
<span id="cb9-42"><a href="#cb9-42" tabindex="-1"></a><span class="co">#&gt;     IWE_rate: 0.005</span></span>
<span id="cb9-43"><a href="#cb9-43" tabindex="-1"></a><span class="co">#&gt;     SSE_evol: function (dt, testing = FALSE) </span></span>
<span id="cb9-44"><a href="#cb9-44" tabindex="-1"></a><span class="co">#&gt;     interval_evol: function (interval_length, dt, testing = FALSE) </span></span>
<span id="cb9-45"><a href="#cb9-45" tabindex="-1"></a><span class="co">#&gt;     mu: 0.005</span></span>
<span id="cb9-46"><a href="#cb9-46" tabindex="-1"></a><span class="co">#&gt;     name: NULL</span></span>
<span id="cb9-47"><a href="#cb9-47" tabindex="-1"></a><span class="co">#&gt;     offspring_index: NULL</span></span>
<span id="cb9-48"><a href="#cb9-48" tabindex="-1"></a><span class="co">#&gt;     own_index: NULL</span></span>
<span id="cb9-49"><a href="#cb9-49" tabindex="-1"></a><span class="co">#&gt;     parent_index: NULL</span></span>
<span id="cb9-50"><a href="#cb9-50" tabindex="-1"></a><span class="co">#&gt;     set_IWE_rate: function () </span></span>
<span id="cb9-51"><a href="#cb9-51" tabindex="-1"></a><span class="co">#&gt;     singleStr: list</span></span>
<span id="cb9-52"><a href="#cb9-52" tabindex="-1"></a><span class="co">#&gt;     singleStr_globalState: M U M</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co"># Example with sampled initial methylation frequencies and default parameter values</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>custom_infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>                             <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>))</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>initialD <span class="ot">&lt;-</span> <span class="fu">simulate_initialData</span>(<span class="at">infoStr =</span> custom_infoStr)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co"># Returns default parameters</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>initialD<span class="sc">$</span>params</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt;   alpha_pI beta_pI alpha_mI beta_mI alpha_pNI beta_pNI alpha_mNI beta_mNI  mu</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; 1      0.1       1      0.1     0.5       0.1        1       0.5      0.1 0.1</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt;   alpha_Ri iota         Ri1         Ri2       Ri3</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt; 1      0.1  0.3 2.80506e-06 0.005770752 0.8942264</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>initialD<span class="sc">$</span>data</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; &lt;combiStructureGenerator&gt;</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt;   Public:</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt;     add_offspring_index: function (i) </span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt;     branch_evol: function (branch_length, dt, testing = FALSE) </span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt;     clone: function (deep = FALSE) </span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt;     copy: function () </span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt;     get_IWE_events: function () </span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="co">#&gt;     get_island_index: function () </span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt;     get_island_number: function () </span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt;     get_mu: function () </span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt;     get_name: function () </span></span>
<span id="cb10-23"><a href="#cb10-23" tabindex="-1"></a><span class="co">#&gt;     get_offspring_index: function () </span></span>
<span id="cb10-24"><a href="#cb10-24" tabindex="-1"></a><span class="co">#&gt;     get_own_index: function () </span></span>
<span id="cb10-25"><a href="#cb10-25" tabindex="-1"></a><span class="co">#&gt;     get_parent_index: function () </span></span>
<span id="cb10-26"><a href="#cb10-26" tabindex="-1"></a><span class="co">#&gt;     get_singleStr: function (i) </span></span>
<span id="cb10-27"><a href="#cb10-27" tabindex="-1"></a><span class="co">#&gt;     get_singleStr_number: function () </span></span>
<span id="cb10-28"><a href="#cb10-28" tabindex="-1"></a><span class="co">#&gt;     initialize: function (infoStr, params = NULL, testing = FALSE) </span></span>
<span id="cb10-29"><a href="#cb10-29" tabindex="-1"></a><span class="co">#&gt;     set_IWE_events: function (a) </span></span>
<span id="cb10-30"><a href="#cb10-30" tabindex="-1"></a><span class="co">#&gt;     set_name: function (a) </span></span>
<span id="cb10-31"><a href="#cb10-31" tabindex="-1"></a><span class="co">#&gt;     set_offspring_index: function (i) </span></span>
<span id="cb10-32"><a href="#cb10-32" tabindex="-1"></a><span class="co">#&gt;     set_own_index: function (i) </span></span>
<span id="cb10-33"><a href="#cb10-33" tabindex="-1"></a><span class="co">#&gt;     set_parent_index: function (i) </span></span>
<span id="cb10-34"><a href="#cb10-34" tabindex="-1"></a><span class="co">#&gt;     set_singleStr: function (singStrList) </span></span>
<span id="cb10-35"><a href="#cb10-35" tabindex="-1"></a><span class="co">#&gt;   Private:</span></span>
<span id="cb10-36"><a href="#cb10-36" tabindex="-1"></a><span class="co">#&gt;     IWE_events: NULL</span></span>
<span id="cb10-37"><a href="#cb10-37" tabindex="-1"></a><span class="co">#&gt;     IWE_rate: 0.1</span></span>
<span id="cb10-38"><a href="#cb10-38" tabindex="-1"></a><span class="co">#&gt;     SSE_evol: function (dt, testing = FALSE) </span></span>
<span id="cb10-39"><a href="#cb10-39" tabindex="-1"></a><span class="co">#&gt;     interval_evol: function (interval_length, dt, testing = FALSE) </span></span>
<span id="cb10-40"><a href="#cb10-40" tabindex="-1"></a><span class="co">#&gt;     mu: 0.1</span></span>
<span id="cb10-41"><a href="#cb10-41" tabindex="-1"></a><span class="co">#&gt;     name: NULL</span></span>
<span id="cb10-42"><a href="#cb10-42" tabindex="-1"></a><span class="co">#&gt;     offspring_index: NULL</span></span>
<span id="cb10-43"><a href="#cb10-43" tabindex="-1"></a><span class="co">#&gt;     own_index: NULL</span></span>
<span id="cb10-44"><a href="#cb10-44" tabindex="-1"></a><span class="co">#&gt;     parent_index: NULL</span></span>
<span id="cb10-45"><a href="#cb10-45" tabindex="-1"></a><span class="co">#&gt;     set_IWE_rate: function () </span></span>
<span id="cb10-46"><a href="#cb10-46" tabindex="-1"></a><span class="co">#&gt;     singleStr: list</span></span>
<span id="cb10-47"><a href="#cb10-47" tabindex="-1"></a><span class="co">#&gt;     singleStr_globalState: M U M</span></span></code></pre></div>
<p>To get the parameter values of a <code>combiStructureGenerator</code>
object, it can alternativelly be given as argument to the function
<code>get_parameterValues()</code>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>combiStr_object <span class="ot">&lt;-</span> initialD<span class="sc">$</span>data</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="fu">get_parameterValues</span>(<span class="at">rootData =</span> combiStr_object)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt;   alpha_pI beta_pI alpha_mI beta_mI alpha_pNI beta_pNI alpha_mNI beta_mNI  mu</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; 1      0.1       1      0.1     0.5       0.1        1       0.5      0.1 0.1</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt;   alpha_Ri iota         Ri1         Ri2       Ri3</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt; 1      0.1  0.3 2.80506e-06 0.005770752 0.8942264</span></span></code></pre></div>
<div id="exploration-of-initial-data" class="section level3">
<h3>Exploration of Initial Data</h3>
<p>The methods of the class <code>combiStructureGenerator</code> can be
used to access each of the contained structures:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="co"># E.g. access fist structure (non-island)</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>combiStr_object<span class="sc">$</span><span class="fu">get_singleStr</span>(<span class="dv">1</span>)</span></code></pre></div>
<p>Each contained structure is an object of
<code>singleStructureGenerator</code> and its methods can be used to get
the contained information.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="co"># E.g. get the methylation equilibrium frequencies of the first singleStructureGenerator</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>combiStr_object<span class="sc">$</span><span class="fu">get_singleStr</span>(<span class="dv">1</span>)<span class="sc">$</span><span class="fu">get_eqFreqs</span>()</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="co">#&gt; [1] 4.802500e-03 3.837392e-05 9.951591e-01</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="co"># E.g. get the sequence of methylation states of the second singleStructureGenerator</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a><span class="co"># Encoded as: 1 for unmethylated, 2 for partially-methylated, 3 for methylated</span></span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>combiStr_object<span class="sc">$</span><span class="fu">get_singleStr</span>(<span class="dv">2</span>)<span class="sc">$</span><span class="fu">get_seq</span>()</span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a><span class="co">#&gt;   [1] 2 2 2 3 1 2 1 2 1 2 1 2 2 1 2 1 2 2 2 1 2 2 2 1 2 2 1 1 1 2 2 1 1 1 2 1 1</span></span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a><span class="co">#&gt;  [38] 1 2 1 1 2 2 2 1 1 1 1 2 1 2 2 1 1 2 2 2 2 2 1 1 1 1 2 2 1 1 2 1 1 1 1 1 1</span></span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a><span class="co">#&gt;  [75] 2 2 2 1 2 2 2 2 2 1 2 1 1 2 2 1 1 2 1 1 1 1 1 1 1 1</span></span></code></pre></div>
</div>
</div>
<div id="simulation-of-data-evolution" class="section level2">
<h2>Simulation of Data Evolution</h2>
<p>The evolution data is generated with the function
<code>simulate_evolData()</code>. It returns the parameters used
(<code>$params</code>), the length of the time step used for the
Gillespie’s <span class="math inline">\(\tau\)</span>-leap approximation
(<code>$dt</code>, default 0.01), the tree used (<code>$tree</code>),
and the simulated data (<code>$data</code>). The description of the
simulated data is available in the function’s documentation.</p>
<p>It takes as parameters:</p>
<ul>
<li><p><code>tree</code>: A string representing a tree in Newick format
<span class="citation">(Olsen 1990)</span>.</p></li>
<li><p><code>infoStr</code>: A dataframe with the information of the
number of CpGs corresponding to each island and non-island region
(columns named <code>n</code> and <code>globalState</code>).
Additionally the initial methylation frequencies can be specified adding
the columns <code>u_eqFreq</code>, <code>p_eqFreq</code> and
<code>m_eqFreq</code>.</p></li>
<li><p><code>dt</code>: The length of time steps for Gillespie’s <span class="math inline">\(\tau\)</span>-leap approximation. Numerical,
default is 0.01.</p></li>
<li><p><code>n_rep</code>: The number of replicates. Integer, default is
1.</p></li>
<li><p><code>only_tip</code>: Logical indicating whether to extract data
only for tips (TRUE, default) or to extract the information for all the
tree branches (FALSE).</p></li>
<li><p><code>params</code>: Default NULL for simulation with default
parameter values. When customized parameter values are used the output
dataframe of <code>get_parameterValues()</code> can be modified and
given to the function. E.g:</p></li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Example with customized initial methylation frequencies, customized parameter values </span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a><span class="co"># and default dt (0.01)</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="st">&quot;((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);&quot;</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>custom_infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>                             <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>),</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>                             <span class="at">u_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>),</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>                             <span class="at">p_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a>                             <span class="at">m_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>custom_params <span class="ot">&lt;-</span> <span class="fu">get_parameterValues</span>()</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>custom_params<span class="sc">$</span>mu <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a>evolD <span class="ot">&lt;-</span> <span class="fu">simulate_evolData</span>(<span class="at">infoStr =</span> custom_infoStr, <span class="at">tree =</span> tree, <span class="at">params =</span> custom_params, <span class="at">n_rep =</span> <span class="dv">3</span>, <span class="at">only_tip =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span>
<span id="cb15-13"><a href="#cb15-13" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span>
<span id="cb15-14"><a href="#cb15-14" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="co"># Returns customized parameters, tree used, time step length for SSE process used (dt) </span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a><span class="co"># and simulated data</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>evolD<span class="sc">$</span>data</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>evolD<span class="sc">$</span>dt</span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>evolD<span class="sc">$</span>tree</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>evolD<span class="sc">$</span>params</span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># The simulated data is returned as a list. Each element of the list corresponds to a</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co"># simulation replicate:</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="fu">length</span>(evolD<span class="sc">$</span>data)</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#&gt; [1] 3</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>rep1 <span class="ot">&lt;-</span> evolD<span class="sc">$</span>data[[<span class="dv">1</span>]]</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>rep2 <span class="ot">&lt;-</span> evolD<span class="sc">$</span>data[[<span class="dv">2</span>]]</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>rep3 <span class="ot">&lt;-</span> evolD<span class="sc">$</span>data[[<span class="dv">3</span>]]</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="co"># When only_tip is TRUE, each replicate contains for each tip: its name and the </span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="co"># sequence of methylation states. E.g.</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>rep1_tip1 <span class="ot">&lt;-</span> rep1[[<span class="dv">1</span>]]</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>rep1_tip1<span class="sc">$</span>name</span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;a&quot;</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="co"># Seq is a list with the methylation states of each simulated structure.</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a><span class="co"># In this example, 3 structures: non-island, island and non-island ...</span></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="fu">length</span>(rep1_tip1<span class="sc">$</span>seq)</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt; [1] 3</span></span>
<span id="cb18-10"><a href="#cb18-10" tabindex="-1"></a><span class="co"># Each with 100 CpGs</span></span>
<span id="cb18-11"><a href="#cb18-11" tabindex="-1"></a><span class="fu">length</span>(rep1_tip1<span class="sc">$</span>seq[[<span class="dv">1</span>]])</span>
<span id="cb18-12"><a href="#cb18-12" tabindex="-1"></a><span class="co">#&gt; [1] 100</span></span></code></pre></div>
<p>When simulations are conducted with <code>only_tip = FALSE</code> the
<code>$data</code> output contains more information.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="st">&quot;((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);&quot;</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>custom_infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>                             <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>),</span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>                             <span class="at">u_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>),</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>                             <span class="at">p_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>                             <span class="at">m_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>custom_params <span class="ot">&lt;-</span> <span class="fu">get_parameterValues</span>()</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>custom_params<span class="sc">$</span>mu <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>evolD <span class="ot">&lt;-</span> <span class="fu">simulate_evolData</span>(<span class="at">infoStr =</span> custom_infoStr, <span class="at">tree =</span> tree, <span class="at">params =</span> custom_params, <span class="at">n_rep =</span> <span class="dv">3</span>, <span class="at">only_tip =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a><span class="co">#&gt; Simulating data at root and letting it evolve along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># When only_tip is FALSE. $data output contains the simulated data and the </span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="co"># information on the relationship between tree branches.</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a><span class="fu">names</span>(evolD<span class="sc">$</span>data)</span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;sim_data&quot;     &quot;branchInTree&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># The information of the tree branches can be accessed using $branchInTree. </span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>treeStr <span class="ot">&lt;-</span> evolD<span class="sc">$</span>data<span class="sc">$</span>branchInTree</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co"># It is a list in which each element index represents the index of the branch</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a>root <span class="ot">&lt;-</span> treeStr[[<span class="dv">1</span>]] <span class="co"># First branch (or tree root)</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>b2 <span class="ot">&lt;-</span> treeStr[[<span class="dv">2</span>]] <span class="co"># Second branch </span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co"># ...</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a><span class="co"># Each branch contains information of the offspring and parent indexes.</span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>root<span class="sc">$</span>parent_index <span class="co"># The tree root does not have parent branches</span></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a>root<span class="sc">$</span>offspring_index <span class="co"># The tree root has 3 daughter branches: 2, 5 and 6. </span></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a><span class="co">#&gt; [1] 2 5 6</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>b2<span class="sc">$</span>parent_index <span class="co"># Branch 2 therefore has as parent branch the root.</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a><span class="co">#&gt; [1] 1</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>b2<span class="sc">$</span>offspring_index <span class="co"># Branch 2 also has 2 daughter branches: 3 and 4</span></span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a><span class="co">#&gt; [1] 3 4</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># The simulated data can be accessed using $sim_data.</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>sim_data <span class="ot">&lt;-</span> evolD<span class="sc">$</span>data<span class="sc">$</span>sim_data</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="co"># As before, each list element corresponds to a simulation replicate. </span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a><span class="co"># E.g. replicate 3</span></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>rep3 <span class="ot">&lt;-</span> sim_data[[<span class="dv">3</span>]]</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a><span class="co"># In each replicate, each element of the list corresponds to a tree branch</span></span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a><span class="co"># the indexes correspond to the information in $branchInTree</span></span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>root <span class="ot">&lt;-</span> rep3[[<span class="dv">1</span>]]</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>b2 <span class="ot">&lt;-</span> rep3[[<span class="dv">2</span>]]</span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a>b3 <span class="ot">&lt;-</span> rep3[[<span class="dv">3</span>]]</span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="co"># Each tree branch contains:</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="co"># - the branch name (NULL for tree root and inner nodes and </span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="co"># tip name for the tree tips):</span></span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>root<span class="sc">$</span>name</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>b2<span class="sc">$</span>name</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>b3<span class="sc">$</span>name</span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;a&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="co"># - the information on IWE events that happened in that branch. </span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>root<span class="sc">$</span>IWE <span class="co"># The root always has NULL because its branch length is 0</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="co"># The rest of the branches have FALSE when no IWE happened or a list containing</span></span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="co"># $islands corresponds to the island index(es) to which the event(s) happened </span></span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a><span class="co"># $times corresponds to the branch time in which the event happened.</span></span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>b2<span class="sc">$</span>IWE</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a><span class="co">#&gt; [1] FALSE</span></span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>b3<span class="sc">$</span>IWE</span>
<span id="cb24-10"><a href="#cb24-10" tabindex="-1"></a><span class="co">#&gt; [1] FALSE</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co"># - A list with each element representing the sequence of methylation states of </span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="co"># each simulated island and non-island</span></span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="co"># Encoded as 0 for unmethylated, 0.5 for partially methylated and 1 for methylated</span></span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a>root<span class="sc">$</span>seq[[<span class="dv">1</span>]] <span class="co"># First sequence (non-island)</span></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="co">#&gt;   [1] 1.0 1.0 1.0 1.0 1.0 0.0 1.0 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 1.0 1.0</span></span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a><span class="co">#&gt;  [19] 0.5 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 0.0 1.0 1.0</span></span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a><span class="co">#&gt;  [37] 0.0 1.0 1.0 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 1.0 1.0 0.5 0.0 1.0 0.0</span></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a><span class="co">#&gt;  [55] 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 1.0 1.0</span></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a><span class="co">#&gt;  [73] 1.0 1.0 1.0 0.5 1.0 0.5 1.0 1.0 1.0 1.0 0.5 1.0 1.0 1.0 1.0 0.5 1.0 0.0</span></span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a><span class="co">#&gt;  [91] 0.0 1.0 1.0 1.0 1.0 1.0 0.0 0.5 1.0 1.0</span></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>root<span class="sc">$</span>seq[[<span class="dv">2</span>]] <span class="co"># Second sequence (island)</span></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a><span class="co">#&gt;   [1] 1.0 0.0 0.0 0.0 0.0 0.0 1.0 1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 1.0 0.5</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a><span class="co">#&gt;  [19] 0.0 0.5 0.0 0.5 0.0 0.0 0.0 0.0 0.5 0.0 0.0 0.5 1.0 0.0 0.0 0.0 0.5 0.0</span></span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a><span class="co">#&gt;  [37] 0.0 0.0 0.0 1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.5 0.0 0.0 1.0</span></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a><span class="co">#&gt;  [55] 0.0 0.0 0.0 0.0 0.5 0.0 0.0 0.0 0.0 0.0 0.0 0.5 0.0 0.0 0.0 0.0 0.0 0.0</span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a><span class="co">#&gt;  [73] 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 1.0 0.0 0.0 1.0 0.0 0.0 0.0 0.0 0.0 0.0</span></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a><span class="co">#&gt;  [91] 0.0 0.0 0.0 0.0 0.0 0.5 0.0 0.0 0.0 0.0</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># - A list with each element representing the methylation frequencies for the states: unmethylated, partially-methylated and methylated</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>root<span class="sc">$</span>eqFreqs[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="co"># First structure of root (non-island) equilibrium frequency for unmethylated state</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="co">#&gt; [1] 0.1</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>root<span class="sc">$</span>eqFreqs[[<span class="dv">1</span>]][<span class="dv">2</span>] <span class="co"># First structure of root (non-island) equilibrium frequency for partially-methylated state</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a><span class="co">#&gt; [1] 0.1</span></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>root<span class="sc">$</span>eqFreqs[[<span class="dv">1</span>]][<span class="dv">3</span>] <span class="co"># First structure of root (non-island) equilibrium frequency for methylated state</span></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a><span class="co">#&gt; [1] 0.8</span></span></code></pre></div>
<ul>
<li><code>rootData</code>: Alternative argument to <code>infoStr</code>
(only one can be given). It enables simulating replicates of evolution
along a tree with fixed data at root. When used, it takes the output of
<code>simulate_initialData()$data</code>. If customized parameter values
are to be used, then the customized parameters are given to
<code>simulate_initialData()</code> instead to
<code>simulate_evolData()</code>.</li>
</ul>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Example with customized initial methylation frequencies, customized parameter values and default dt (0.01)</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="st">&quot;((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);&quot;</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>custom_infoStr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n =</span> <span class="fu">c</span>(<span class="dv">100</span>, <span class="dv">100</span>, <span class="dv">100</span>),</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>                             <span class="at">globalState =</span> <span class="fu">c</span>(<span class="st">&quot;M&quot;</span>, <span class="st">&quot;U&quot;</span>, <span class="st">&quot;M&quot;</span>),</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>                             <span class="at">u_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>),</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>                             <span class="at">p_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a>                             <span class="at">m_eqFreq =</span> <span class="fu">c</span>(<span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a>custom_params <span class="ot">&lt;-</span> <span class="fu">get_parameterValues</span>()</span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>custom_params<span class="sc">$</span>mu <span class="ot">&lt;-</span> <span class="fl">0.005</span></span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a>initialD <span class="ot">&lt;-</span> <span class="fu">simulate_initialData</span>(<span class="at">infoStr =</span> custom_infoStr, <span class="at">params =</span> custom_params)<span class="sc">$</span>data</span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a>evolD <span class="ot">&lt;-</span> <span class="fu">simulate_evolData</span>(<span class="at">rootData =</span>initialD, <span class="at">tree =</span> tree)</span>
<span id="cb27-12"><a href="#cb27-12" tabindex="-1"></a><span class="co">#&gt; Parameter values set as in given rootData</span></span>
<span id="cb27-13"><a href="#cb27-13" tabindex="-1"></a><span class="co">#&gt; Simulating evolution of given data at root along given tree:  ((a:1, b:1):2, c:2, (d:3.7, (e:4, f:1):3):5);</span></span></code></pre></div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-R6" class="csl-entry">
Chang, Winston. 2022. <em>R6: Encapsulated Classes with Reference
Semantics</em>.
</div>
<div id="ref-gillespie2001approximate" class="csl-entry">
Gillespie, Daniel T. 2001. <span>“Approximate Accelerated Stochastic
Simulation of Chemically Reacting Systems.”</span> <em>The Journal of
Chemical Physics</em> 115 (4): 1716–33.
</div>
<div id="ref-olsen1990interpretation" class="csl-entry">
Olsen, Gary. 1990. <span>“Interpretation of the &quot;Newick’s 8: 45&quot; Tree
Format Standard.”</span> <a href="https://phylipweb.github.io/phylip/newick_doc.html">https://phylipweb.github.io/phylip/newick_doc.html</a>.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
