<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Summary Statistics</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Summary Statistics</h1>


<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#summary-statistics">Summary Statistics</a>
<ul>
<li><a href="#frequency-of-methylation-states-per-genomic-structure-type">Frequency of Methylation States per Genomic Structure Type</a>
<ul>
<li><a href="#global-mean">Global Mean</a></li>
<li><a href="#mean-sd-of-observed-frequencies-per-structure-type-at-tree-tips">Mean SD of Observed Frequencies per Structure Type at Tree Tips</a></li>
</ul></li>
<li><a href="#mean-correlation-of-a-central-segment-of-methylation-states-per-genomic-structure-type">Mean Correlation of a Central Segment of Methylation States per Genomic Structure Type</a></li>
<li><a href="#comparison-of-tree-tips">Comparison of tree tips</a>
<ul>
<li><a href="#mean-frequency-of-methylation-state-changes-at-single-sites-per-structure-type">Mean Frequency of Methylation State Changes at Single Sites per Structure Type</a></li>
<li><a href="#fitch-parsimony-estimate-for-minimum-global-methylation-state-changes-at-cpg-islands">Fitch Parsimony Estimate for Minimum Global Methylation State Changes at CpG Islands</a></li>
<li><a href="#mean-number-of-significant-methylation-frequency-changes-per-island-and-cherry">Mean Number of Significant Methylation Frequency Changes per Island and Cherry</a></li>
<li><a href="#mean-number-of-significant-methylation-frequency-changes-per-island-across-all-tree-tips">Mean Number of Significant Methylation Frequency Changes per Island Across All Tree Tips</a></li>
</ul></li>
</ul></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(MethEvolSIM)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here we present summary statistics that can be used, e.g. in an Approximate Bayesian Computation method <span class="citation">(Csillery, Francois, and Blum 2012)</span>, to estimate parameters of the methylation dynamics model of MethEvolSIM. The statistics include mean frequencies, standard deviations, neighbor correlations of methylation states in island and non-island regions. Additionally, for analyses comparing different samples (comparison of tree tips), some of the summary statistics are based on the mean per-site frequency of methylation changes and the minimum number of changes per islands leading to different global methylation states, calculated with the Fitch algorithm <span class="citation">(Fitch 1971)</span>.</p>
<p>The provided set of functions allows the computation of summary statistics for methylation data in a genomic region with structural categorization into two types of structures. In this vignette, we will refer to these structures as <strong>islands</strong> and <strong>non-island structures</strong>.</p>
<p>The <code>data</code> argument must be in one of the following formats, depending on the analysis:</p>
<ol style="list-style-type: decimal">
<li><strong>For analyses without tip comparison:</strong> A list where each element represents the methylation states in a given structure (<code>data[[structure]]</code>):</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="co"># Example: a single sample with 3 genomic structures</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a><span class="co"># (1) island with 10 partially-methylated sites</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a><span class="co"># (2) non-island with 5 methylated sites</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true"></a><span class="co"># (3) island with 15 unmethylated sites</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>, <span class="dv">10</span>),  <span class="co"># Partially methylated</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true"></a>             <span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">5</span>),      <span class="co"># Methylated</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true"></a>             <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">15</span>))     <span class="co"># Unmethylated</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true"></a>data</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true"></a><span class="co">#&gt; [1] 1 1 1 1 1</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true"></a><span class="co">#&gt;  [1] 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li><strong>For analyses comparing multiple tips in a phylogenetic tree:</strong> A nested list where the first level represents the tips, and the second level represents the structures within each tip (<code>data[[tip]][[structure]]</code>).</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="co"># Example: data from 3 tips of a tree,</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="co"># each with 3 genomic structures</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a>    <span class="co"># Tip 1</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>)),  <span class="co"># 5 partially methylated, 5 unmethylated</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">4</span>), <span class="fl">0.5</span>),         <span class="co"># 4 methylated, 1 unmethylated</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">7</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">8</span>))), <span class="co"># 7 unmethylated, 8 partially methylated</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true"></a>    <span class="co"># Tip 2</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># 9 partially methylated, 1 methylated</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">4</span>), <span class="dv">1</span>), <span class="co"># 4 partially methylated, 1 methylated</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">7</span>))), <span class="co"># 8 unmethylated, 7 partially-methylated</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true"></a>    <span class="co"># Tip 3</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>), <span class="dv">1</span>), <span class="co"># first and last methylated, rest unmethylated</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">3</span>), <span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">2</span>)), <span class="co"># 3 methylated, 1 unmethylated</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">15</span>)))) <span class="co"># all partially methylated</span></span></code></pre></div>
<p>In this case, <code>data</code> must be pre-filtered to include only sites present in all tips, ensuring a valid comparison between species or samples.</p>
<p>Regardless of the format, methylation values should be represented as follows:</p>
<ul>
<li><p><code>0</code> for unmethylated sites,</p></li>
<li><p><code>0.5</code> for partially methylated sites,</p></li>
<li><p><code>1</code> for methylated sites.</p></li>
</ul>
<p>Intermediate values (e.g., obtained from pooled empirical data) should be categorized before analysis with <code>categorize_siteMethSt</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a>non_categorized_data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a>  <span class="co"># Tip 1</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.7</span>, <span class="fl">0.9</span>), <span class="co"># First structure</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.9</span>)), <span class="co"># Second structure</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a>  <span class="co"># Tip 2</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.8</span>, <span class="fl">0.6</span>), <span class="co"># First structure</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="fl">0.9</span>, <span class="fl">0.4</span>, <span class="fl">0.7</span>)) <span class="co"># Second structure</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a>  )</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true"></a>  </span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true"></a><span class="co"># Transform the data with custom thresholds</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true"></a>categorized_data &lt;-<span class="st"> </span><span class="kw">categorize_siteMethSt</span>(data, <span class="dt">u_threshold =</span> <span class="fl">0.15</span>, <span class="dt">m_threshold =</span> <span class="fl">0.85</span>)</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true"></a></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true"></a>categorized_data</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true"></a><span class="co">#&gt; [[1]][[1]]</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.5 0.5 0.5 0.5 0.5 0.0 0.0 0.0 0.0 0.0</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true"></a><span class="co">#&gt; [[1]][[2]]</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true"></a><span class="co">#&gt; [1] 1.0 1.0 1.0 1.0 0.5</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true"></a><span class="co">#&gt; [[1]][[3]]</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true"></a><span class="co">#&gt; [[2]][[1]]</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 1.0</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true"></a><span class="co">#&gt; [[2]][[2]]</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true"></a><span class="co">#&gt; [1] 0.5 0.5 0.5 0.5 1.0</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true"></a><span class="co">#&gt; [[2]][[3]]</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.5 0.5 0.5 0.5 0.5 0.5 0.5</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true"></a><span class="co">#&gt; [[3]][[1]]</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true"></a><span class="co">#&gt;  [1] 1 0 0 0 0 0 0 0 0 1</span></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true"></a><span class="co">#&gt; [[3]][[2]]</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true"></a><span class="co">#&gt; [1] 0.5 0.5 0.5 1.0 1.0</span></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true"></a><span class="co">#&gt; [[3]][[3]]</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true"></a><span class="co">#&gt;  [1] 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5 0.5</span></span></code></pre></div>
<p>Finally, when phylogenetic data is provided, the order of tips in <code>data</code> must match the tip order in the Newick tree format (left to right) or the order of <code>tree$tip.label()</code> when the tree is provided as a <code>phylo</code> object from the <code>ape</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="co"># Example tree in Newick format for the above data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>newick_tree &lt;-<span class="st"> &quot;((tip1:1, tip2:1):1, tip3:2);&quot;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a><span class="co"># Example tree as a phylo object from the ape package</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a><span class="kw">library</span>(ape)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a>phylo_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> newick_tree)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a>phylo_tree<span class="op">$</span>tip.label</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a><span class="co">#&gt; [1] &quot;tip1&quot; &quot;tip2&quot; &quot;tip3&quot;</span></span></code></pre></div>
</div>
<div id="summary-statistics" class="section level1">
<h1>Summary Statistics</h1>
<div id="frequency-of-methylation-states-per-genomic-structure-type" class="section level2">
<h2>Frequency of Methylation States per Genomic Structure Type</h2>
<div id="global-mean" class="section level3">
<h3>Global Mean</h3>
<ul>
<li><strong>Partially-methylated sites</strong>: Let <span class="math inline">\(x_{i,t}\)</span> and <span class="math inline">\(y_{j,t}\)</span> be the frequency of methylation state <span class="math inline">\(p\)</span> in island <span class="math inline">\(i \in\{1, \cdots, I\}\)</span> and non-island <span class="math inline">\(j \in\{1, \cdots, J\}\)</span> at tip <span class="math inline">\(t\)</span>, respectively. We define the mean global frequency of <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> as:</li>
</ul>
<p><span class="math display">\[
\overline{x}=\frac1I\sum_{i=1}^{I}\frac1T\sum_{t=1}^{T}x_{i,t} \\
\overline{y}=\frac1J\sum_{j=1}^{J}\frac1T\sum_{t=1}^{T}y_{j,t}
\]</span></p>
<p>where <span class="math inline">\(x_{i,t}\)</span> and <span class="math inline">\(y_{j,t}\)</span> represent the respective frequencies at each tip.</p>
<ul>
<li><strong>Methylated sites</strong>: Similarly, let <span class="math inline">\(w_{i,t}\)</span> and <span class="math inline">\(z_{j,t}\)</span> be the frequency of methylation state <span class="math inline">\(m\)</span> in islands and non-islands. The mean global frequencies are:</li>
</ul>
<p><span class="math display">\[
\overline{w}=\frac1I\sum_{i=1}^{I}\frac1T\sum_{t=1}^{T}w_{i,t} \\
\overline{z}=\frac1J\sum_{j=1}^{J}\frac1T\sum_{t=1}^{T}z_{j,t}
\]</span></p>
<p>These values can be computed using the following functions:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a><span class="co"># 1 tip / sample / replicate</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">.5</span>)) <span class="co"># tip 1</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true"></a><span class="kw">get_islandMeanFreqP</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true"></a><span class="co">#&gt; [1] 0.5833333</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true"></a><span class="kw">get_nonislandMeanFreqP</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true"></a><span class="co">#&gt; [1] 0.4166667</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true"></a><span class="kw">get_islandMeanFreqM</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true"></a><span class="kw">get_nonislandMeanFreqM</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1666667</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a><span class="co"># 2 tip / sample / replicate</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>)), <span class="co"># tip 1</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">1</span>)) <span class="co"># tip 2</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true"></a>  )</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true"></a><span class="kw">get_islandMeanFreqP</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true"></a><span class="co">#&gt; [1] 0.5833333</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true"></a><span class="kw">get_nonislandMeanFreqP</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true"></a><span class="co">#&gt; [1] 0.4166667</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true"></a><span class="kw">get_islandMeanFreqM</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true"></a><span class="co">#&gt; [1] 0.08333333</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true"></a><span class="kw">get_nonislandMeanFreqM</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true"></a><span class="co">#&gt; [1] 0.2916667</span></span></code></pre></div>
<p>Note that in the calculation of the means the relative frequencies <span class="math inline">\(x_{i, t}\)</span>, <span class="math inline">\(y_{j, t}\)</span>, <span class="math inline">\(w_{i, t}\)</span> and <span class="math inline">\(z_{j, t}\)</span> are <strong>not</strong> weighted by the lengths of island <span class="math inline">\(i\)</span> or non-island <span class="math inline">\(j\)</span>.</p>
</div>
<div id="mean-sd-of-observed-frequencies-per-structure-type-at-tree-tips" class="section level3">
<h3>Mean SD of Observed Frequencies per Structure Type at Tree Tips</h3>
<ul>
<li><strong>Partially-methylated sites</strong>: We define the standard deviation of <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span> at tip <span class="math inline">\(t\)</span> as:</li>
</ul>
<p><span class="math display">\[
\sigma_{t}(x)=\sqrt{\frac{1}{I-1}\sum_{i=1}^{I}(x_{i,t}-\overline{x_{.,t}})^2} \\
\sigma_{t}(y)=\sqrt{\frac{1}{J-1}\sum_{j=1}^{J}(y_{j,t}-\overline{y_{.,t}})^2}
\]</span></p>
<p>where <span class="math inline">\(\overline{x_{.,t}}\)</span> and <span class="math inline">\(\overline{y_{.,t}}\)</span> represent mean frequencies at tip <span class="math inline">\(t\)</span>. The mean standard deviation across tips is:</p>
<p><span class="math display">\[
\hat{\sigma}(x)=\frac{1}{T}\sum_{t=1}^{T}\sigma_t(x) \\
\hat{\sigma}(y)=\frac{1}{T}\sum_{t=1}^{T}\sigma_t(y)
\]</span></p>
<ul>
<li><strong>Methylated sites</strong>: Similarly, for <span class="math inline">\(w\)</span> and <span class="math inline">\(z\)</span>:</li>
</ul>
<p><span class="math display">\[
\hat{\sigma}(w)=\frac{1}{T}\sum_{t=1}^{T}\sigma_t(w) \\
\hat{\sigma}(z)=\frac{1}{T}\sum_{t=1}^{T}\sigma_t(z)
\]</span></p>
<p>These values can be computed using:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true"></a><span class="co"># 1 tip / sample / replicate</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>))</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true"></a><span class="kw">get_islandSDFreqP</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1178511</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true"></a><span class="kw">get_nonislandSDFreqP</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1178511</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true"></a><span class="kw">get_islandSDFreqM</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true"></a><span class="co">#&gt; [1] 0.2357023</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true"></a><span class="kw">get_nonislandSDFreqM</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1767767</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a><span class="co"># 2 tip / sample / replicate</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>)), <span class="co"># tip 1</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">1</span>)) <span class="co"># tip 2</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true"></a>  )</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true"></a><span class="kw">get_islandSDFreqP</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true"></a><span class="co">#&gt; [1] 0.05892557</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true"></a><span class="kw">get_nonislandSDFreqP</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1178511</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true"></a><span class="kw">get_islandSDFreqM</span>(index_islands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true"></a><span class="co">#&gt; [1] 0.1767767</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true"></a><span class="kw">get_nonislandSDFreqM</span>(index_nonislands, data, <span class="dt">categorized_data =</span> T, sample_n)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true"></a><span class="co">#&gt; [1] 0.05892557</span></span></code></pre></div>
<p>Note that we average across tips as there can be correlations between the mean standard deviation of tips (e.g. under the case of an event affecting the states at two tips).</p>
</div>
</div>
<div id="mean-correlation-of-a-central-segment-of-methylation-states-per-genomic-structure-type" class="section level2">
<h2>Mean Correlation of a Central Segment of Methylation States per Genomic Structure Type</h2>
<p>Let <span class="math inline">\(s\)</span> be the length of two adjacent segments of central positions to consider. Let <span class="math inline">\(l\)</span> be the number of positions excluded at the start and end of each structure. Define:</p>
<p><span class="math display">\[
\overline{Cor(x)_{s,l}}=\frac1I\frac1T\sum_{i=1}^{I}\sum_{t=1}^{T}Cor(x_{i,t,[p:p&#39;-1]}, x_{i,t,[p+1:p&#39;]}) \\
\overline{Cor(y)_{s,l}}=\frac1J\frac1T\sum_{j=1}^{J}\sum_{t=1}^{T}Cor(y_{j,t,[p:p&#39;-1]}, y_{j,t,[p+1:p&#39;]})
\]</span></p>
<p>Correlations are computed only for segment pairs where standard deviation is non-zero:</p>
<p><span class="math display">\[
\sigma(x_{i,t,[p:p&#39;-1]}) \neq 0 \text{ and } \sigma(x_{i,t,[p+1:p&#39;]})\neq 0 \\
\sigma(y_{j,t,[p:p&#39;-1]}) \neq 0 \text{ and } \sigma(y_{j,t,[p+1:p&#39;]})\neq 0
\]</span></p>
<p>In this formulation:</p>
<ul>
<li><p>Only structures with a minimum length of <span class="math inline">\(n = s + 2 \cdot l\)</span> are considered.</p></li>
<li><p>If <span class="math inline">\(l = 0\)</span>, all the positions are considered.</p></li>
</ul>
<p>These values can be computed with the functions <code>compute_meanCor_i</code> for island structures and <code>compute_meanCor_ni</code> for non-island structures as in the following examples:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true"></a><span class="co"># 1 tip / sample / replicate</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">1</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>) </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true"></a>               <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>), <span class="co"># 30 sites</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true"></a>               <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true"></a>                 <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="co"># 25 sites</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true"></a>               <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true"></a>                 <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true"></a>                 <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>), <span class="co"># 40 sites</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true"></a>               <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true"></a>                 <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true"></a>                 <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>)) <span class="co"># 35 sites</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true"></a><span class="kw">compute_meanCor_i</span>(index_islands, <span class="dt">minN_CpG =</span> <span class="dv">10</span>, </span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true"></a>                  <span class="dt">shore_length =</span> <span class="dv">5</span>, data, <span class="dt">sample_n =</span> <span class="dv">1</span>, <span class="dt">categorized_data =</span> T)</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true"></a><span class="co">#&gt; [1] 0.3626687</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true"></a><span class="kw">compute_meanCor_ni</span>(index_nonislands, <span class="dt">minN_CpG =</span> <span class="dv">10</span>, </span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true"></a>                   <span class="dt">shore_length =</span> <span class="dv">5</span>, data, <span class="dt">sample_n =</span> <span class="dv">1</span>, <span class="dt">categorized_data =</span> T)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true"></a><span class="co">#&gt; [1] 0.3623932</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true"></a><span class="co"># 2 tip / sample / replicate</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true"></a>sample_n &lt;-<span class="st"> </span><span class="dv">2</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">4</span>)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true"></a>  <span class="co"># tip 1</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, </span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true"></a>           <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>), <span class="co"># 30 sites</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true"></a>         <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="co"># 25 sites</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true"></a>         <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, </span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, </span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">1</span>), <span class="co"># 40 sites</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, </span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true"></a>           <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>)), <span class="co"># 35 sites</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true"></a>  <span class="co"># tip 2</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="co"># 30 sites</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true"></a>         <span class="kw">c</span>(.<span class="dv">5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>), <span class="co"># 25 sites</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true"></a>         <span class="kw">c</span>(.<span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true"></a>           <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, </span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true"></a>           <span class="fl">.5</span>, <span class="dv">1</span>), <span class="co"># 40 sites</span></span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="dv">1</span>, </span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true"></a>           <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="fl">.5</span>, <span class="dv">0</span>, <span class="fl">.5</span>)) <span class="co"># 35 sites</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true"></a>  )</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true"></a><span class="kw">compute_meanCor_i</span>(index_islands, <span class="dt">minN_CpG =</span> <span class="dv">10</span>, </span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true"></a>                  <span class="dt">shore_length =</span> <span class="dv">5</span>, data, <span class="dt">sample_n =</span> <span class="dv">2</span>, <span class="dt">categorized_data =</span> T)</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true"></a><span class="co">#&gt; [1] 0.4371918</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true"></a><span class="kw">compute_meanCor_ni</span>(index_nonislands, <span class="dt">minN_CpG =</span> <span class="dv">10</span>, </span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true"></a>                   <span class="dt">shore_length =</span> <span class="dv">5</span>, data, <span class="dt">sample_n =</span> <span class="dv">2</span>, <span class="dt">categorized_data =</span> T)</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true"></a><span class="co">#&gt; [1] 0.5021673</span></span></code></pre></div>
<p>Note that correlation measures how much the state of each site says about the next. So, when there is no variation (e.g. in middle segment state u of a site is followed by state u of the next), there is no correlation.</p>
</div>
<div id="comparison-of-tree-tips" class="section level2">
<h2>Comparison of tree tips</h2>
<div id="mean-frequency-of-methylation-state-changes-at-single-sites-per-structure-type" class="section level3">
<h3>Mean Frequency of Methylation State Changes at Single Sites per Structure Type</h3>
<p>Let <span class="math inline">\(S_k\)</span> be the set of sites in the genomic structure indexed by <span class="math inline">\(k\)</span>. For each site <span class="math inline">\(s \in S_k\)</span>, let the methylation state at a given tip <span class="math inline">\(t\)</span> be denoted as <span class="math inline">\(m_{k,s,t}\)</span>, where <span class="math inline">\(m_{k,s,t} \in \{0, 0.5, 1\}\)</span> represents the unmethylated, partially-methylated, or methylated state, respectively.</p>
<p>For a given cherry <span class="math inline">\((t_1, t_2)\)</span>, that is two tips <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span> that are direct offspring of the same internal node, we compute the proportion of sites with differing methylation states between the two tips.</p>
<p>Define the indicator variable: <span class="math display">\[
\delta_{k,s,t_1,t_2} =
\begin{cases}
1, &amp; \text{if } m_{k,s,t_1} \neq m_{k,s,t_2}, \\
0, &amp; \text{if } m_{k,s,t_1} = m_{k,s,t_2}.
\end{cases}
\]</span></p>
<p>Then, the proportion of sites with different methylation states in structure <span class="math inline">\(k\)</span> for the cherry <span class="math inline">\((t_1, t_2)\)</span> is given by:</p>
<p><span class="math display">\[
F_k(t_1, t_2) = \frac{1}{|S_k|} \sum_{s \in S_k} \delta_{k,s,t_1,t_2}
\]</span></p>
<p>where <span class="math inline">\(|S_k|\)</span> is the total number of sites in structure <span class="math inline">\(k\)</span>, and the sum counts the number of sites where the methylation state differs between tips <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span>.</p>
<p>This statistic quantifies the proportion of differing methylation states for a given genomic structure in a cherry.</p>
<p>To compute the weighted mean frequency of methylation changes across all structures in a cherry, we separate island and non-island structures.</p>
<p>For a given cherry <span class="math inline">\((t_1, t_2)\)</span>, let <span class="math inline">\(\mathcal{I}\)</span> be the set of indices for island structures and <span class="math inline">\(\mathcal{N}\)</span> be the set of indices for non-island structures. Each structure <span class="math inline">\(k\)</span> has a total number of sites <span class="math inline">\(|S_k|\)</span>, which we use as weights.</p>
<p>The weighted mean frequency of methylation changes for island structures is:</p>
<p><span class="math display">\[
\bar{F}_{\text{island}}(t_1, t_2) = \frac{\sum_{k \in \mathcal{I}} |S_k| F_k(t_1, t_2)}{\sum_{k \in \mathcal{I}} |S_k|}
\]</span></p>
<p>Similarly, the weighted mean frequency for non-island structures is:</p>
<p><span class="math display">\[
\bar{F}_{\text{non-island}}(t_1, t_2) = \frac{\sum_{k \in \mathcal{N}} |S_k| F_k(t_1, t_2)}{\sum_{k \in \mathcal{N}} |S_k|}
\]</span></p>
<p>These expressions compute the mean frequency of methylation state differences across all island and non-island structures, weighted by the number of sites in each structure.</p>
<p>Functions:</p>
<ul>
<li><p><code>get_cherryDist(tree)</code> to get the distances between the tips of each cherry. It identifies each cherry by the tip names and the tip indices. The tip indices correspond to (a) the index from left to right on the newick string, (b) the order of the tip label in the <code>phylo_object$tip.label</code>, and (c) the index in the methylation data list <code>data[[tip]][[structure]]</code> as obtained with the function <code>simulate_evolData()</code> when the given tree has several tips.</p></li>
<li><p><code>countSites_cherryMethDiff(cherryDist, data)</code> to get for each cherry the counts of sites with half-methylation and full-methylation changes per genomic structure (island or non-island).</p></li>
<li><p><code>freqSites_cherryMethDiff(tree,data)</code>. The function first validates the tree structure and extracts pairwise distances between cherry tips with <code>get_cherryDist</code>. It then validates the input data and counts half and full methylation differences for each cherry at each structure using <code>countSites_cherryMethDiff</code>. Finally, it normalizes these (half and full) counts by the number of sites per structure to compute frequencies.</p></li>
<li><p><code>get_siteFChange_cherry(tree,data)</code> uses <code>freqSites_cherryMethDiff(tree,data)</code> and then computes the frequency of sites with any change of methylation state (full or half) for each cherry at each genomic structure.</p></li>
<li><p><code>MeanSiteFChange_cherry(tree, data, index_islands, index_nonislands)</code>. The function computes the per-cherry frequency of sites with different methylation states at each structure (islands and non-islands) using <code>get_siteFChange_cherry</code>. Then it calculates the weighted mean site frequency of methylation changes for each cherry separately for islands and non-islands.</p></li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true"></a><span class="co"># Set example tree and methylation data</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true"></a>tree &lt;-<span class="st"> &quot;((a:1.5,b:1.5):2,(c:2,d:2):1.5);&quot;</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">10</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">8</span>)), <span class="co"># tip a</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">10</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>)), <span class="co"># tip b</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">10</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>)), <span class="co"># tip c</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>, <span class="dv">5</span>)), <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>), <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="dv">1</span>, <span class="kw">rep</span>(<span class="dv">0</span>, <span class="dv">6</span>)))) <span class="co"># d </span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true"></a><span class="co"># Set the index for islands and non-island structures</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">2</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true"></a>index_nonislands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true"></a><span class="kw">MeanSiteFChange_cherry</span>(<span class="dt">data =</span> data, </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true"></a>                       <span class="dt">categorized_data =</span> T, </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true"></a>                       <span class="dt">tree =</span> tree, </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true"></a>                       <span class="dt">index_islands =</span> index_islands, </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true"></a>                       <span class="dt">index_nonislands =</span> index_nonislands)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true"></a><span class="co">#&gt;   tip_names tip_indices dist nonisland_meanFChange island_meanFChange</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true"></a><span class="co">#&gt; 1       a-b         1-2    3             0.4444444                  1</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true"></a><span class="co">#&gt; 2       c-d         3-4    4             0.6666667                  1</span></span></code></pre></div>
</div>
<div id="fitch-parsimony-estimate-for-minimum-global-methylation-state-changes-at-cpg-islands" class="section level3">
<h3>Fitch Parsimony Estimate for Minimum Global Methylation State Changes at CpG Islands</h3>
<p>Let <span class="math inline">\(I\)</span> be the set of CpG islands in a given genomic region For each island <span class="math inline">\(i \in I\)</span>, the mean methylation level across sites is computed with <code>get_meanMeth_islands(index_islands, data)</code> and categorized into one of three global methylation states—unmethylated (<span class="math inline">\(u\)</span>), partially methylated (<span class="math inline">\(p\)</span>), or methylated (<span class="math inline">\(m\)</span>)—based on user-defined thresholds <span class="math inline">\(u_{\text{thresh}}\)</span> and <span class="math inline">\(m_{\text{thresh}}\)</span>. This categorization is given by the function:</p>
<p><span class="math display">\[
\operatorname{categorize\_islandGlbSt}(\mu_i, u_{\text{thresh}}, m_{\text{thresh}}) =
\begin{cases}
  u, &amp; \text{if } \mu_i \leq u_{\text{thresh}}, \\
  m, &amp; \text{if } \mu_i \geq m_{\text{thresh}}, \\
  p, &amp; \text{otherwise}.
\end{cases}
\]</span></p>
<p>Given a phylogenetic tree <span class="math inline">\(T\)</span> with tips corresponding to sampled species or individuals, we aim to estimate the minimum number of state changes required to explain the observed distribution of island methylation states across the tips. This is achieved using the Fitch algorithm <span class="citation">(Fitch 1971)</span> as implemented in <code>compute_fitch(meth, tree)</code>.</p>
<p>The function <code>computeFitch_islandGlbSt(T, M)</code>, where <span class="math inline">\(M\)</span> is the matrix of categorized methylation states across tips, returns a vector <span class="math inline">\(C\)</span> where each entry <span class="math inline">\(C_i\)</span> represents the minimum number of changes required for island <span class="math inline">\(i\)</span> under the Fitch parsimony criterion.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true"></a><span class="co"># Example with data from a single island structure </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true"></a><span class="co"># and three tips</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true"></a>tree &lt;-<span class="st"> &quot;((bla:1,bah:1):2,booh:2);&quot;</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true"></a>  <span class="co">#Tip 1</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>))), <span class="co"># m</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true"></a>  <span class="co">#Tip 2</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">10</span>))), <span class="co"># p</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true"></a>  <span class="co">#Tip 3</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)))) <span class="co"># p</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true"></a>  </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>)</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true"></a>  </span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true"></a><span class="kw">computeFitch_islandGlbSt</span>(index_islands, data, tree, </span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true"></a>                         <span class="dt">u_threshold =</span> <span class="fl">0.1</span>, <span class="dt">m_threshold =</span> <span class="fl">0.9</span>)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true"></a><span class="co">#&gt; [1] 1</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true"></a><span class="co"># Example: data from a genomic region consisting on 3 structures with 10 sites each</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true"></a><span class="co"># one island, one non-island, one island</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true"></a><span class="co"># and a tree with 8 tips</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true"></a>tree &lt;-<span class="st"> &quot;(((a:1,b:1):1,(c:1,d:1):1):1,((e:1,f:1):1,(g:1,h:1):1):1);&quot;</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true"></a>data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true"></a>  <span class="co">#Tip 1</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>)), <span class="co"># p</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true"></a>        <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># m</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true"></a>  <span class="co">#Tip 2</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># p</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true"></a>        <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># u</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true"></a>  <span class="co">#Tip 3</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># m</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># p</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true"></a>  <span class="co">#Tip 4</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># m</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="dv">0</span>), </span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># p</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true"></a>  <span class="co">#Tip 5</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">5</span>)), <span class="co"># u</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># p</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true"></a>  <span class="co">#Tip 6</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># u</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># m</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true"></a>  <span class="co">#Tip 7</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># u</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), </span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))), <span class="co"># u</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true"></a>  <span class="co">#Tip 8</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true"></a>  <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># u</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="dv">0</span>), </span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true"></a>       <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)))) <span class="co"># u</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true"></a>  </span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true"></a>index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true"></a>  </span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true"></a><span class="kw">computeFitch_islandGlbSt</span>(index_islands, data, tree, </span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true"></a>                         <span class="dt">u_threshold =</span> <span class="fl">0.1</span>, <span class="dt">m_threshold =</span> <span class="fl">0.9</span>)</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true"></a><span class="co">#&gt; [1] 2 4</span></span></code></pre></div>
<p>For genomic regions containing multiple CpG islands, we can summarize the per-island estimates using the mean to obtain an overall measure of the minimum state changes.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true"></a><span class="kw">mean</span>(<span class="kw">computeFitch_islandGlbSt</span>(index_islands, data, tree, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true"></a>                              <span class="dt">u_threshold =</span> <span class="fl">0.1</span>, <span class="dt">m_threshold =</span> <span class="fl">0.9</span>))</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true"></a><span class="co">#&gt; [1] 3</span></span></code></pre></div>
</div>
<div id="mean-number-of-significant-methylation-frequency-changes-per-island-and-cherry" class="section level3">
<h3>Mean Number of Significant Methylation Frequency Changes per Island and Cherry</h3>
<p>A cherry <span class="math inline">\(c\)</span> is defined as a pair of direct offspring (tips <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span>) of the same internal node in the phylogenetic tree. For a given cherry <span class="math inline">\(c\)</span>, let <span class="math inline">\(\mathcal{I}\)</span> be the set of indices for island structures. For each island structure <span class="math inline">\(i \in \mathcal{I}\)</span> at each cherry tip <span class="math inline">\(t_c \in \{t_{c,1}, t_{c,2}\}\)</span> in each phylogenetic tree, the number of sites in each methylation state (unmethylated <span class="math inline">\(0\)</span>, partially-methylated <span class="math inline">\(0.5\)</span>, methylated <span class="math inline">\(1\)</span>) is counted. This count is represented as:</p>
<p><span class="math display">\[
\text{count}_{\text{UPM}}(c,t_c, i) = \left( n_0, n_{0.5}, n_1 \right)
\]</span></p>
<p>where <span class="math inline">\(n_0\)</span>, <span class="math inline">\(n_{0.5}\)</span>, and <span class="math inline">\(n_1\)</span> are the counts of unmethylated, partially-methylated, and methylated sites at tip <span class="math inline">\(t_c\)</span>, respectively.</p>
<p>The distribution of methylation states at each island between the two tips <span class="math inline">\(t_1\)</span> and <span class="math inline">\(t_2\)</span> of a given cherry is compared using a chi-squared test. The null hypothesis of the test is that the frequency distributions of states in the two islands follow the same multinomial distribution. Note that not only IWEs can lead to deviations from this null hypothesis but also neighbor-dependent SSEs can lead to slight deviations from the assumption of multinomial distributions. The test statistic is calculated using the contingency table <span class="math inline">\(T_{c,i}\)</span> for tips <span class="math inline">\(t_{c,1}\)</span> and <span class="math inline">\(t_{c,2}\)</span>:</p>
<p><span class="math display">\[
T_{c,i} = \begin{pmatrix}
n_0^{t_{c,1}} &amp; n_{0.5}^{t_{c,1}} &amp; n_1^{t_{c,1}} \\
n_0^{t_{c,2}} &amp; n_{0.5}^{t_{c,2}} &amp; n_1^{t_{c,2}}
\end{pmatrix}
\]</span> For each cherry <span class="math inline">\(c\)</span> and each island <span class="math inline">\(i\)</span>, the methylation frequencies at each cherry tip are compared using the chi-squared test. The p-value for the chi-squared test is calculated via Monte Carlo simulation to improve reliability, even when the expected frequencies do not meet the assumptions of the chi-squared approximation (i.e., expected counts of at least 5 in each category). The obtained p-values are stored in a dataframe for each cherry and for each island.</p>
<p>The significance of the methylation frequency changes is determined based on a user-defined threshold <span class="math inline">\(p_{\text{threshold}}\)</span>. If the p-value is smaller than the threshold, the change is considered significant. This is done for each cherry and each island <span class="math inline">\(i \in \mathcal{I}\)</span> by comparing each p-value to the threshold:</p>
<p><span class="math display">\[
\text{Significant Change}_{c,i} = 
\begin{cases} 
1, &amp; \text{if } p_{\text{value}}(c,i) \lt p_{\text{threshold}} \\
0, &amp; \text{if } p_{\text{value}}(c,i) \ge p_{\text{threshold}}
\end{cases}
\]</span></p>
<p>For each cherry , the mean number of significant changes across all islands <span class="math inline">\(i \in \mathcal{I}\)</span> is computed as the average of the significant changes for each island:</p>
<p><span class="math display">\[
\text{Mean Number of Significant Changes per Island}_{c}
= \frac{1}{|\mathcal{I}|} \sum_{i \in \mathcal{I}} \text{Significant Change}_{c,i}
\]</span></p>
<p>where <span class="math inline">\(|\mathcal{I}|\)</span> is the total number of islands.</p>
<p>Functions:</p>
<ul>
<li><p><code>compare_CherryFreqs(tip1, tip2)</code> to perform a chi-squared test to compare the distribution of methylation states between two cherry tips.</p></li>
<li><p><code>pValue_CherryFreqsChange_i(data, index_islands, tree)</code> uses <code>compare_CherryFreqs(tip1, tip2)</code> to get for each cherry and each island the chi-square p-value of the comparison of the distribution of methylation states between two cherry tips.</p></li>
<li><p><code>mean_CherryFreqsChange_i(data, categorized_data = FALSE, index_islands, tree, pValue_threshold)</code>. The function uses <code>pValue_CherryFreqsChange_i(data, index_islands, tree)</code> and counts the mean number of significant changes per island at each cherry.</p></li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true"></a><span class="co"># Set example tree and methylation data</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true"></a>  tree &lt;-<span class="st"> &quot;((a:1,b:1):2,c:2);&quot;</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true"></a>  data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true"></a>    <span class="co">#Tip a</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>)), <span class="co"># Structure 1: island</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>))),  <span class="co"># Structure 3: island</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true"></a>    <span class="co">#Tip b</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)),  <span class="co"># Structure 1: island</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>))), <span class="co"># Structure 3: island</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true"></a>    <span class="co">#Tip c</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)),  <span class="co"># Structure 1: island</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)))) <span class="co"># Structure 3: island</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true"></a>  </span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true"></a>  </span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true"></a>  index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)</span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true"></a>  </span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true"></a>  <span class="kw">mean_CherryFreqsChange_i</span>(data, <span class="dt">categorized_data =</span> T,</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true"></a>                           index_islands, tree,</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true"></a>                           <span class="dt">pValue_threshold =</span> <span class="fl">0.05</span>)</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true"></a><span class="co">#&gt;   first_tip_name second_tip_name first_tip_index second_tip_index dist island_1</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true"></a><span class="co">#&gt; 1              a               b               1                2    2     TRUE</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true"></a><span class="co">#&gt;   island_3 FreqsChange</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true"></a><span class="co">#&gt; 1    FALSE         0.5</span></span></code></pre></div>
</div>
<div id="mean-number-of-significant-methylation-frequency-changes-per-island-across-all-tree-tips" class="section level3">
<h3>Mean Number of Significant Methylation Frequency Changes per Island Across All Tree Tips</h3>
<p>For a given phylogenetic tree with a number of tips <span class="math inline">\(N\)</span>, let <span class="math inline">\(t_n \in \{t_1, t_2, \cdots, t_N\)</span> represent the tree tips. For a given genomic region, let <span class="math inline">\(\mathcal{I}\)</span> be the set of indices for island structures. For each island structure <span class="math inline">\(i \in \mathcal{I}\)</span> at each tip <span class="math inline">\(t_n\)</span> in the phylogenetic tree, the number of sites in each methylation state (unmethylated <span class="math inline">\(0\)</span>, partially-methylated <span class="math inline">\(0.5\)</span>, methylated <span class="math inline">\(1\)</span>) is counted. This count is represented as:</p>
<p><span class="math display">\[
\text{count}_{\text{UPM}}(t_n, i) = \left( n_0, n_{0.5}, n_1 \right)
\]</span></p>
<p>where <span class="math inline">\(n_0\)</span>, <span class="math inline">\(n_{0.5}\)</span>, and <span class="math inline">\(n_1\)</span> are the counts of unmethylated, partially-methylated, and methylated sites at tip <span class="math inline">\(t_n\)</span>, respectively.</p>
<p>The distribution of methylation states at each island across all tips is compared using a chi-squared test. The null hypothesis of the test is that the frequency distributions of states in the two islands follow the same multinomial distribution. Note that not only IWEs can lead to deviations from this null hypothesis but also neighbor-dependent SSEs can lead to slight deviations from the assumption of multinomial distributions. The test statistic is calculated using the contingency table <span class="math inline">\(T_{i}\)</span> for all tree tips:</p>
<p><span class="math display">\[
T_{i} = \begin{pmatrix}
n_0^{t_{1}} &amp; n_{0.5}^{t_{1}} &amp; n_1^{t_{1}} \\
n_0^{t_{2}} &amp; n_{0.5}^{t_{2}} &amp; n_1^{t_{2}} \\
\cdots &amp; \cdots &amp; \cdots \\
n_0^{t_{N}} &amp; n_{0.5}^{t_{N}} &amp; n_1^{t_{N}} 
\end{pmatrix}
\]</span></p>
<p>For each island <span class="math inline">\(i\)</span>, the methylation frequencies are compared across tips using the chi-squared test. The p-value for the chi-squared test is calculated via Monte Carlo simulation to improve reliability, even when the expected frequencies do not meet the assumptions of the chi-squared approximation (i.e., expected counts of at least 5 in each category).</p>
<p>The significance of the methylation frequency changes for each island <span class="math inline">\(i \in \mathcal{I}\)</span> across tips is determined based on a user-defined threshold <span class="math inline">\(p_{\text{threshold}}\)</span>. If the p-value is smaller than the threshold, the change is considered significant.</p>
<p><span class="math display">\[
\text{Significant Change}_{i} = 
\begin{cases} 
1, &amp; \text{if } p_{\text{value}}(i) &lt; p_{\text{threshold}} \\
0, &amp; \text{if } p_{\text{value}}(i) \ge p_{\text{threshold}}
\end{cases}
\]</span></p>
<p>For the given tree, the mean number of significant changes per island across all tips is computed as:</p>
<p><span class="math display">\[
\text{Mean Number of Significant Changes per Island}
= \frac{1}{|\mathcal{I}|} \sum_{i \in \mathcal{I}} \text{Significant Change}_{i}
\]</span></p>
<p>where <span class="math inline">\(|\mathcal{I}|\)</span> is the total number of islands.</p>
<p>Function:</p>
<ul>
<li><code>mean_TreeFreqsChange_i(tree, data, categorized_data = FALSE, index_islands, pValue_threshold)</code>.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true"></a><span class="co"># Set example tree and methylation data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true"></a>tree &lt;-<span class="st"> &quot;((a:1,b:1):2,(c:2,d:2):1);&quot;</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true"></a>  data &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true"></a>    <span class="co">#Tip a</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>)), <span class="co"># Structure 1: island</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>))), <span class="co"># Structure 3: island</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true"></a>    <span class="co">#Tip b</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># Structure 1: island</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>))),<span class="co"># Structure 3: island</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true"></a>    <span class="co">#Tip c</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># Structure 1: island</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">1</span>))),<span class="co"># Structure 3: island</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true"></a>    <span class="co">#Tip d</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true"></a>    <span class="kw">list</span>(<span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">9</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">1</span>)), <span class="co"># Structure 1: island</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">9</span>), <span class="dv">1</span>), <span class="co"># Structure 2: non-island</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true"></a>         <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">1</span>,<span class="dv">8</span>), <span class="kw">rep</span>(<span class="fl">0.5</span>,<span class="dv">2</span>)))) <span class="co"># Structure 3: island</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true"></a>  </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true"></a>  </span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true"></a>  index_islands &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true"></a>  </span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true"></a>  <span class="kw">mean_TreeFreqsChange_i</span>(tree, data, <span class="dt">categorized_data =</span> T,</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true"></a>                           index_islands,</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true"></a>                           <span class="dt">pValue_threshold =</span> <span class="fl">0.05</span>)</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true"></a><span class="co">#&gt; [1] 1</span></span></code></pre></div>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references hanging-indent">
<div id="ref-csillery2012abc">
<p>Csillery, Katalin, Olivier Francois, and Michael G. B. Blum. 2012. “Abc: An R Package for Approximate Bayesian Computation (Abc).” <em>Methods in Ecology and Evolution</em>.</p>
</div>
<div id="ref-fitch1971toward">
<p>Fitch, Walter M. 1971. “Toward Defining the Course of Evolution: Minimum Change for a Specific Tree Topology.” <em>Systematic Biology</em> 20 (4): 406–16.</p>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
